modules/govuk_postgresql/manifests/server/not_slave.pp

```diff
@@ -3,10 +3,20 @@
 # Collect virtual resources that should exist on `server::standalone` and
 # `server::primary` nodes but NOT `server::standby`.
 #
-class govuk_postgresql::server::not_slave {
-  Postgresql::Server::Role           <| tag == $name |>
-  Govuk_postgresql::Rds_sql          <| tag == $name |>
-  Postgresql::Server::Db             <| tag == $name |>
-  Postgresql::Server::Database_grant <| tag == $name |>
-  Govuk_postgresql::Extension        <| tag == $name |>
+# === Parameters
+#
+# [*ensure*]
+#   Enables collecting of virtual resources if present
+#
+class govuk_postgresql::server::not_slave(
+  $ensure = present,
+) {
+
+  if $ensure == present {
+    Postgresql::Server::Role           <| tag == $name |>
+    Govuk_postgresql::Rds_sql          <| tag == $name |>
+    Postgresql::Server::Db             <| tag == $name |>
+    Postgresql::Server::Database_grant <| tag == $name |>
+    Govuk_postgresql::Extension        <| tag == $name |>
+  }
```

Comment:
I don't think we should have an ensure here since it seems to just be queries this does (I think) rather than install software.